pkg/dashboard/resource/functiontemplate.go

```diff
@@ -56,16 +76,110 @@ func (ftr *functionTemplateResource) GetAll(request *http.Request) (map[string]r
 
-		// add to attributes
-		attributes[matchingFunctionTemplate.Name] = restful.Attributes{
-			"metadata": matchingFunctionTemplate.Configuration.Meta,
-			"spec":     matchingFunctionTemplate.Configuration.Spec,
+		// if not rendered, add template in "values" mode, else just add as functionConfig with Meta and Spec
+		if matchingFunctionTemplate.FunctionConfigTemplate != "" {
+			var values map[string]interface{}
+			err := yaml.Unmarshal([]byte(matchingFunctionTemplate.FunctionConfigValues), &values)
+			if err != nil {
+				return nil, errors.Wrap(err, "Failed to unmarshall function template's values file")
+			}
+
+			attributes[matchingFunctionTemplate.Name] = restful.Attributes{
+				"template": matchingFunctionTemplate.FunctionConfigTemplate,
+				"values":   values,
+			}
+		} else {
+
+			// add to attributes
+			attributes[matchingFunctionTemplate.Name] = restful.Attributes{
+				"metadata": matchingFunctionTemplate.FunctionConfig.Meta,
+				"spec":     matchingFunctionTemplate.FunctionConfig.Spec,
+			}
 		}
 	}
 
 	return attributes, nil
 }
 
+// returns a list of custom routes for the resource
+func (ftr *functionTemplateResource) GetCustomRoutes() ([]restful.CustomRoute, error) {
+
+	// since delete and update by default assume /resource/{id} and we want to get the id/namespace from the body
+	// we need to register custom routes
+	return []restful.CustomRoute{
+		{
+			Pattern:   "/render",
+			Method:    http.MethodPost,
+			RouteFunc: ftr.Render,
+		},
+	}, nil
+}
+
+func getTemplateFileWithValues (templateFile string, values map[string]interface{}) string {
+	for valueName, valueInterface := range values {
+		templateFile = strings.Replace(templateFile, "{{ ." + valueName + " }}", valueInterface.(string), -1)
+	}
+	return templateFile
+}
+
+func getFunctionConfigFromTemplate (templateFile string) (*functionconfig.Config, error) {
+	funcitonConfig := functionconfig.Config{}
+
+	err := json.Unmarshal([]byte(templateFile), &funcitonConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to unmarshall function config from template file")
+	}
+
+	return &funcitonConfig, nil
+}
+
+func (ftr *functionTemplateResource) resourceToAttributes(resource interface{}) restful.Attributes {
+
+	s := structs.New(resource)
+
+	// use "json" tag to specify how to serialize the keys
+	s.TagName = "json"
+
+	return s.Map()
+}
+
+func (ftr *functionTemplateResource) Render(request *http.Request) (*restful.CustomRouteFuncResponse, error) {
+
+	statusCode := http.StatusNoContent
+
+	// read body
+	body, err := ioutil.ReadAll(request.Body)
+	if err != nil {
+		return nil, nuclio.WrapErrInternalServerError(errors.Wrap(err, "Failed to read body"))
+	}
+
+	renderGivenValues := RenderConfig{}
+	err = json.Unmarshal(body, &renderGivenValues)
+	if err != nil {
+		return nil, nuclio.WrapErrBadRequest(errors.Wrap(err, "Failed to parse JSON body"))
+	}
+
+	// from values to template
```

Comment:
i think I lost you here... You simply need to use https://golang.org/pkg/text/template - take template + values = functionconfig